be/src/vec/common/typeid_cast.h

```diff
@@ -43,34 +43,20 @@
   */
 template <typename To, typename From>
     requires std::is_reference_v<To>
-To typeid_cast(From& from) {
-    try {
-        if (typeid(from) == typeid(To)) {
-            return static_cast<To>(from);
-        }
-    } catch (const std::exception& e) {
-        throw doris::Exception(doris::ErrorCode::BAD_CAST, e.what());
-    }
+To typeid_cast(From& from) noexcept(false) {
+    if ((typeid(From) == typeid(To)) || (typeid(from) == typeid(To))) return static_cast<To>(from);
 
     throw doris::Exception(doris::ErrorCode::BAD_CAST,
                            "Bad cast from type " + demangle(typeid(from).name()) + " to " +
                                    demangle(typeid(To).name()));
 }
 
 template <typename To, typename From>
-To typeid_cast(From* from) {
-#ifndef NDEBUG
-    try {
-        if (typeid(*from) == typeid(std::remove_pointer_t<To>)) {
-            return static_cast<To>(from);
-        }
-    } catch (const std::exception& e) {
-        throw doris::Exception(doris::ErrorCode::BAD_CAST, e.what());
-    }
-#else
-    if (typeid(*from) == typeid(std::remove_pointer_t<To>)) {
+    requires std::is_pointer_v<To>
+To typeid_cast(From* from) noexcept {
```

Comment:
In debug mode, the previous code will print exception. It is useful to help us debug.

Reply:
I dont know whether `typeid_cast(nullptr)` is designed as a common use scenario